### ISSUE TITLE
Implement audio file purging

### ISSUE THREAD

**Issue by tphakala** [tphakala/birdnet-go#69]
Old wav files should be purged once disk space is starting to get low, purging should be flexible so that users can select which species recordings should be deleted last.

**Comment by isZumpo** [comment]
Should only the audio file be removed or the whole database entry for a given detection?

**Comment by MueJosh** [comment]
> Should only the audio file be removed or the whole database entry for a given detection?


Definitely just the audio files. (It makes much more sense this way)

**Comment by isZumpo** [comment]
> > Should only the audio file be removed or the whole database entry for a given detection?
> 
> Definitely just the audio files. (It makes much more sense this way)

Agreed! The database entries should be of so little size that it does not make any sense to clear them up in comparison to the clips. They could also be useful for long term statistics and insights.

**Comment by isZumpo** [comment]
I got started on a commit where I made a system that would check and make sure that we don't go above 90% disk usage for the given filesystem where the clips are saved. Deleting the oldest clips to always keep it below for example 90%. However, I just kept thinking of edge cases that makes it very hard to not end up in unfortunate situations.  For instance. what if the user decides to put a large blob of data that by itself fills up 90% of the filesystem, then the clips use lets say 1% of the filesystem. Should we then just go ahead and delete all bird clips to keep it below 90%? Feels like there are so many ways to lose important clips that way...

Instead of diving into complicated disk usage algorithms to not end up in these situations above, I thought we instead could come up with some simpler retention rules. One simple idea to get started that would be good enough for a first policy would be this:

We add these two variables to the config:
**MinEvictionHours** - Minimum time before recording is considered for eviction
**MinRecordingsPerSpecies** - Number of recordings to always keep per species

Then every x minutes or on new detections, we would query the database for detections older than **MinEvictionHours** where there are a minimum of **MinRecordingsPerSpecies** for that species. Then delete all clips for those, maybe nulling the database entry for the detection itself?

Does this sound like a reasonable first attempt on implementing some sort of retention strategy? Of course in the future we could consider things as disk utilization, and confidence, rarity of the detection as parameters to some policy.

**Comment by tphakala** [comment]
Audio file purging has been implemented
